### PR TITLE
Rename Shifter to Gateway, shiftIn/Out to mint/burn

### PR DESCRIPTION
**Pull request by noiach**
**Description**

The following names have been updated:

* Shifter -> Gateway
* ShifterRegistry -> GatewayRegistry
* ShiftedERC20 -> RenERC20
  * zBTC -> RenBTC, etc.
* shiftIn -> mint
* shiftOut -> burn
* shiftOutRef -> n

### CODE REVIEW

File: build/chaosnet/BCHShifter.json [hunk 1]
```diff
@@ -562,7 +562,7 @@
       "type": "function"
     }
   ],
-  "sourcePath": "/home/noah/github/renproject/darknode-sol/contracts/Shifter/Shifter.sol",
+  "sourcePath": "./contracts/Shifter/Shifter.sol",
```
Comment (loongy) [inline build/chaosnet/BCHShifter.json]: These build artefacts are still using the old names.

File: build/chaosnet/BasicAdapter.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
+{
+  "contractName": "BasicAdapter",
+  "abi": [
+    {
+      "constant": false,
+      "inputs": [
+        {
+          "internalType": "contract IShifterRegistry",
```
Comment (loongy) [inline build/chaosnet/BasicAdapter.json]: Still using old naming scheme.

File: build/chaosnet/zBTC.json [hunk 3]
```diff
@@ -431,7 +431,7 @@
       "type": "function"
     }
   ],
-  "sourcePath": "/home/noah/github/renproject/darknode-sol/contracts/Shifter/ERC20Shifted.sol",
+  "sourcePath": "./contracts/Shifter/ERC20Shifted.sol",
```
Comment (loongy) [inline build/chaosnet/zBTC.json]: Still using old naming scheme.

File: build/chaosnet/BCHShifter.json (hunk 1 above)
Comment (noiach) [reply build/chaosnet/BCHShifter.json]: The build files will be updated as they are deployed.

File: build/chaosnet/BasicAdapter.json (hunk 2 above)
Comment (noiach) [reply build/chaosnet/BasicAdapter.json]: Same as above.

File: build/chaosnet/zBTC.json (hunk 3 above)
Comment (noiach) [reply build/chaosnet/zBTC.json]: Same as above.

**Pull request merged**